File: POO/guia02/Notas.cpp
#include <iostream>
#include <string>

using namespace std;

typedef struct notas {
  float notaU1;
  float notaU2;
  float notaP;
} notas;

class Alunos {
private:
  notas notasAluno[3];
  string Nome;
  int notaF;

  float notaUnidade(int unidade) {
    return notasAluno[unidade].notaU1 * 0.2 + notasAluno[unidade].notaU2 * 0.2 +
           notasAluno[unidade].notaP * 0.6;
  }

public:
  Alunos(string nome, float notasU1[], float notasU2[], float notasP[])
      : Nome(nome) {
    for (int i = 0; i < 3; ++i) {
      notasAluno[i].notaU1 = notasU1[i];
      notasAluno[i].notaU2 = notasU2[i];
      notasAluno[i].notaP = notasP[i];
    }
    notaF = 0;
  };

  float notaFinal() {
    if (notaF == 0) {
      for (int i = 0; i < 3; ++i)
        notaF += notaUnidade(i);
      notaF /= 3;
    }
    return notaF;
  }
  string getNome() { return Nome; }
};

int main() {
  float notasU1[3] = {
      1.2, 1.3, 1.4}; // nota da primeira guia de laboratorio de cada unidade
  float notaU2[3] = {2, 2,
                     2}; // nota da segunda guia de laboratorio de cada unidade
  float notaP[3] = {5, 5, 5};
  string nome = "danilo";

  Alunos umALuno = Alunos(nome, notasU1, notaU2, notaP);
  cout << "nome: " << umALuno.getNome()
       << ", nota final: " << umALuno.notaFinal() << endl;
}

File: periodo_2/POO/guia02/Notas.java
public class Notas {

	private String nome;
    private float[] notasU1;
    private float[] notasU2;
    private float[] notasP;
    private float notaF;

	public Notas(String nome, float[] notasU1, float[] notasU2, float[] notasP) {
	    this.nome = nome;    
        this.notasU1 = notasU1;
        this.notasU2 = notasU2;
        this.notasP = notasP;
	}
	public String getNome() {
		return nome;
	}
    private float notaUnidade(int unidade) {
        return (this.notasU1[unidade] + this.notasU2[unidade]) * 0.2f +
            this.notasP[unidade] * 0.6f;
    }
    public float notaFinal() {
        if (notaF == 0) {
          for (int i = 0; i < 3; ++i)
            notaF += notaUnidade(i);
          notaF /= 3;
        }
        return notaF;
    }
	public static void main(String[] args) {
        float[] notasU1 = {1,2,3},
                notasU2 = {1,2,3},
                notasP  = {1,2,3};
		Notas umALuno = new Notas("danilo", notasU1, notasU2, notasP);
		System.out.print("nome: "+ umALuno.getNome()+ ", notaF: " + umALuno.notaFinal());
	}
};
